### ISSUE TITLE
SELECT LOWER not work

### ISSUE THREAD

**Issue by liubaohai** [pingcap/tidb#19393]
### 1. Minimal reproduce step (Required)

- create table sql
```
MySQL [test]> show create table testhex;
+---------+---------------------------------------------------------------------------------------------------------------+
| Table   | Create Table                                                                                                  |
+---------+---------------------------------------------------------------------------------------------------------------+
| testhex | CREATE TABLE `testhex` (
  `hex_id` binary(16) NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |
+---------+---------------------------------------------------------------------------------------------------------------+
```
- insert into a row
```
 insert into testhex VALUES (UNHEX("aabbcc"));
```
- select
```
MySQL [test]> SELECT LOWER(HEX(`hex_id`)) as `hex_id` FROM `testhex`;
+----------------------------------+
| hex_id                           |
+----------------------------------+
| AABBCC00000000000000000000000000 |
+----------------------------------+
```

### 2. What did you expect to see? (Required)

```
MySQL [test]> SELECT LOWER(HEX(`hex_id`)) as `hex_id` FROM `testhex`;
+----------------------------------+
| hex_id                           |
+----------------------------------+
| aabbcc00000000000000000000000000 |
+----------------------------------+
```

### 4. What is your TiDB version? (Required)

```
tiup cluster deploy tidb-test v4.0.0 ./topology.yaml --user root
```

```
MySQL [test]> select * from mysql.tidb where VARIABLE_NAME ='tidb_server_version';
+---------------------+----------------+-----------------------------------+
| VARIABLE_NAME       | VARIABLE_VALUE | COMMENT                           |
+---------------------+----------------+-----------------------------------+
| tidb_server_version | 46             | Bootstrap version. Do not delete. |
+---------------------+----------------+-----------------------------------+
1 row in set (0.01 sec)
```

**Comment by fzhedu** [comment]
Thanks for your first issue. Would you like to solve it ?

**Comment by fzhedu** [comment]
This is fixed in the master and release 4.0, but still exist on v4.0.0, also fixed since 4.0.2. please upgrade your version.

**Comment by fzhedu** [comment]
This is fixed in the master and release 4.0, but still exist on v4.0.0, also fixed since 4.0.2. please upgrade your version.

so I close it.

**Comment by fzhedu** [comment]
https://github.com/pingcap/tidb/pull/14667 fixed PR.
